### ISSUE TITLE
[request] npoco Attributes in a seperate assembly

### ISSUE THREAD

**Issue by euioa** [schotime/NPoco#318]
Hello,

I am working on an N-tiers application. In my scenario the dto's are in a separate assembly witch is shared between the _client app_ and the _server app_. The client app dont connect directly to the db.

dto's are decorated with npoco attributes (PrimaryKeyAttribute, TableNameAttribute, SerializedColumnAttribute...). 

To properly implement this kind of scenario it would be great if we can deploy npoco attributes in a separate assembly from Nuget. Thus the DAL will not be deployed on the client.

To summarize :
It should be 2 deployment modes.
1- Full : for "server apps" or "smart client apps who connect to the DB" ; We deploy the _main Npoco assembly_ and the _Npoco attributes assembly_.  (which is equivalent to today situation)

2- Client : for a "not connected Tier" : We only deploy the _Npoco attributes assembly_.

Thanks

**Comment by imasm** [comment]
Have you tried Fluent Mapping?

https://github.com/schotime/NPoco/wiki/Fluent-mappings

**Comment by euioa** [comment]
Thank you for the answer.
I know about Fluent mapping but I prefer to use attributes.
In any case, I believe There are no disadvantages to cut the npoco dll in two.
